code/modules/mob/living/silicon/laws.dm

```diff
@@ -11,13 +14,17 @@
 	deadchat_broadcast("'s <b>laws were changed.</b> <a href='?src=[REF(src)]&dead=1&printlawtext=[url_encode(lawtext)]'>View</a>", "<span class='name'>[src]</span>", follow_target=src, message_type=DEADCHAT_LAWCHANGE)
 
 /mob/living/silicon/proc/post_lawchange(announce = TRUE)
+	if(last_lawchange == world.time)
+		return FALSE
```

Comment:
@Rohesie It won't run more than once per tick however procs dealing with entire lawsets need to wait for all individual law changes before running.